src/main/java/com/uber/jenkins/phabricator/conduit/DifferentialClient.java

```diff
@@ -0,0 +1,122 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.uber.jenkins.phabricator.conduit;
+
+import com.uber.jenkins.phabricator.LauncherFactory;
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * DifferentialClient handles all interaction with conduit/arc for differentials
+ */
+public class DifferentialClient {
+    private final String arcPath;
+    private final String conduitToken;
+    private final LauncherFactory launcher;
+    private final String diffID;
+
+    public DifferentialClient(String diffID, LauncherFactory launcher, String conduitToken, String arcPath) {
+        this.diffID = diffID;
+        this.launcher = launcher;
+        this.conduitToken = conduitToken;
+        this.arcPath = arcPath;
+    }
+
+    /**
+     * Posts a comment to a differential
+     * @param message
+     * @param silent whether or not to trigger an email
+     * @param action phabricator comment action, e.g. 'resign', 'reject', 'none'
+     */
+    public JSONObject postComment(String message, boolean silent, String action) throws IOException, InterruptedException, ArcanistUsageException {
+        Map params = new HashMap<String, String>();
+        params.put("revision_id", this.diffID);
+        params.put("action", action);
+        params.put("message", message);
+        params.put("silent", silent);
+
+        return this.callConduit("differential.createcomment", params);
+    }
+
+    public JSONObject fetchDiff() throws ArcanistUsageException, IOException, InterruptedException {
+        Map params = new HashMap<String, String>();
+        params.put("ids", new String[]{this.diffID});
+        JSONObject query = this.callConduit("differential.querydiffs", params);
+        JSONObject response;
+        try {
+            response = query.getJSONObject("response");
+        } catch (JSONException e) {
+            e.printStackTrace();
```

Comment:
Hmm... won't this just go to the jenkins process' stdout/stderr? That won't be that useful